### PR TITLE
Add confirmation dialog when collected data will be discarded

### PR DESCRIPTION
**Pull request by JSunde**

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt [hunk 1]
```diff
@@ -120,7 +121,20 @@ abstract class AbstractTaskFragment<T : AbstractTaskViewModel> : AbstractFragmen
 
   protected fun addSkipButton() =
     addButton(ButtonAction.SKIP)
-      .setOnClickListener { onSkip() }
+      .setOnClickListener {
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt:124]: Can we extract this event handler to a fun?

**Review by gino-m — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskFragment.kt]: Resolved, went with onSkip()

**Comment by JSunde** [comment]
> @JSunde I'm unable to test this feature on HEAD. Can you please double check that the dialog is visible for you when attempting to skip? Also, can we add one unit test for it?

Did you try skipping after data had been input? 

And yea I can look into adding a unit test for this.

**Comment by shobhitagarwal1612** [comment]
Yes, I tried doing it with select type tasks

**Comment by JSunde** [comment]
Oh huh, you're right. I think I had tested with the multiple choice task previously but not after changing the implementation slightly. It seems to work properly with the Text input task.

I'm guessing it's an issue with the way I'm checking the response text to see if the user has input any data.

**Pull request merged**